<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.550">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Courtney Schiffman">
<meta name="author" content="Michel Friesenhahn">
<meta name="author" content="Christina Rabe">
<meta name="dcterms.date" content="2022-10-12">

<title>stats4datascience - How to get the most out of prognostic baseline variables in clinical trials</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../assets/favicon.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../assets/styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../assets/favicon.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">stats<span class="fancy-four">4</span>datascience</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">How to get the most out of prognostic baseline variables in clinical trials</h1>
            <p class="subtitle lead">Effective strategies for employing covariate adjustment and stratification</p>
                                <div class="quarto-categories">
                <div class="quarto-category">covariate adjustment</div>
                <div class="quarto-category">stratification</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Courtney Schiffman </p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://www.gene.com/">
              Genentech
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Michel Friesenhahn </p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://www.gene.com/">
              Genentech
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Christina Rabe </p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://www.gene.com/">
              Genentech
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 12, 2022</p>
      </div>
    </div>
    
      <div>
      <div class="quarto-title-meta-heading">Modified</div>
      <div class="quarto-title-meta-contents">
        <p class="date-modified">February 21, 2024</p>
      </div>
    </div>
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#new-insights-into-an-old-method" id="toc-new-insights-into-an-old-method" class="nav-link active" data-scroll-target="#new-insights-into-an-old-method">New Insights into an Old Method</a></li>
  <li><a href="#covariate-adjustment-has-gotten-hip" id="toc-covariate-adjustment-has-gotten-hip" class="nav-link" data-scroll-target="#covariate-adjustment-has-gotten-hip">Covariate Adjustment Has Gotten Hip</a></li>
  <li><a href="#whats-the-big-deal-about-covariate-adjustment" id="toc-whats-the-big-deal-about-covariate-adjustment" class="nav-link" data-scroll-target="#whats-the-big-deal-about-covariate-adjustment">What’s the Big Deal About Covariate Adjustment?</a>
  <ul class="collapse">
  <li><a href="#covariate-adjustment-increases-precision-and-power" id="toc-covariate-adjustment-increases-precision-and-power" class="nav-link" data-scroll-target="#covariate-adjustment-increases-precision-and-power">Covariate Adjustment Increases Precision and Power</a></li>
  <li><a href="#covariate-adjustment-removes-the-impact-of-imbalance" id="toc-covariate-adjustment-removes-the-impact-of-imbalance" class="nav-link" data-scroll-target="#covariate-adjustment-removes-the-impact-of-imbalance">Covariate Adjustment Removes the Impact of Imbalance</a></li>
  <li><a href="#covariate-adjustment-is-rigorous" id="toc-covariate-adjustment-is-rigorous" class="nav-link" data-scroll-target="#covariate-adjustment-is-rigorous">Covariate Adjustment is Rigorous</a>
  <ul class="collapse">
  <li><a href="#sampling-framework" id="toc-sampling-framework" class="nav-link" data-scroll-target="#sampling-framework">Sampling framework</a></li>
  <li><a href="#target-estimand" id="toc-target-estimand" class="nav-link" data-scroll-target="#target-estimand">Target Estimand</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#conditional-average-treatment-effect" id="toc-conditional-average-treatment-effect" class="nav-link" data-scroll-target="#conditional-average-treatment-effect">Conditional Average Treatment Effect</a></li>
  <li><a href="#marginal-average-treatment-effect" id="toc-marginal-average-treatment-effect" class="nav-link" data-scroll-target="#marginal-average-treatment-effect">Marginal Average Treatment Effect</a></li>
  <li><a href="#assumptions" id="toc-assumptions" class="nav-link" data-scroll-target="#assumptions">Assumptions</a></li>
  <li><a href="#covariate-adjustment-is-encouraged-by-the-fda" id="toc-covariate-adjustment-is-encouraged-by-the-fda" class="nav-link" data-scroll-target="#covariate-adjustment-is-encouraged-by-the-fda">Covariate Adjustment is Encouraged by the FDA</a></li>
  <li><a href="#how-to-perform-covariate-adjustment" id="toc-how-to-perform-covariate-adjustment" class="nav-link" data-scroll-target="#how-to-perform-covariate-adjustment">How to Perform Covariate Adjustment</a>
  <ul class="collapse">
  <li><a href="#standardized-regression-estimator" id="toc-standardized-regression-estimator" class="nav-link" data-scroll-target="#standardized-regression-estimator">Standardized Regression Estimator</a></li>
  <li><a href="#inference-for-tams-and-ates" id="toc-inference-for-tams-and-ates" class="nav-link" data-scroll-target="#inference-for-tams-and-ates">Inference for TAMs and ATEs</a></li>
  <li><a href="#determining-your-model-budget" id="toc-determining-your-model-budget" class="nav-link" data-scroll-target="#determining-your-model-budget">Determining your Model Budget</a></li>
  </ul></li>
  <li><a href="#model-budget" id="toc-model-budget" class="nav-link" data-scroll-target="#model-budget">Model Budget</a></li>
  <li><a href="#spending-your-model-budget-wisely" id="toc-spending-your-model-budget-wisely" class="nav-link" data-scroll-target="#spending-your-model-budget-wisely">Spending your Model Budget Wisely</a></li>
  <li><a href="#accounting-for-stratified-randomization" id="toc-accounting-for-stratified-randomization" class="nav-link" data-scroll-target="#accounting-for-stratified-randomization">Accounting for Stratified Randomization</a></li>
  <li><a href="#performance-metrics" id="toc-performance-metrics" class="nav-link" data-scroll-target="#performance-metrics">Performance Metrics</a>
  <ul class="collapse">
  <li><a href="#intuitive-metrics-for-quantifying-the-expected-benefit-of-adjustment" id="toc-intuitive-metrics-for-quantifying-the-expected-benefit-of-adjustment" class="nav-link" data-scroll-target="#intuitive-metrics-for-quantifying-the-expected-benefit-of-adjustment">Intuitive Metrics for Quantifying the Expected Benefit of Adjustment</a></li>
  <li><a href="#essi-formulas" id="toc-essi-formulas" class="nav-link" data-scroll-target="#essi-formulas">ESSI Formulas</a></li>
  </ul></li>
  <li><a href="#general-essi" id="toc-general-essi" class="nav-link" data-scroll-target="#general-essi">General ESSI</a></li>
  <li><a href="#constant-absolute-treatment-effect" id="toc-constant-absolute-treatment-effect" class="nav-link" data-scroll-target="#constant-absolute-treatment-effect">Constant absolute treatment effect</a></li>
  <li><a href="#constant-absolute-treatment-effect-1" id="toc-constant-absolute-treatment-effect-1" class="nav-link" data-scroll-target="#constant-absolute-treatment-effect-1">Constant absolute treatment effect:</a></li>
  <li><a href="#constant-proportional-treatment-effect" id="toc-constant-proportional-treatment-effect" class="nav-link" data-scroll-target="#constant-proportional-treatment-effect">Constant proportional treatment effect</a></li>
  <li><a href="#constant-proportional-treatment-effect-1" id="toc-constant-proportional-treatment-effect-1" class="nav-link" data-scroll-target="#constant-proportional-treatment-effect-1">Constant proportional treatment effect</a></li>
  <li><a href="#impact-of-htes-on-gains-from-covariate-adjustment" id="toc-impact-of-htes-on-gains-from-covariate-adjustment" class="nav-link" data-scroll-target="#impact-of-htes-on-gains-from-covariate-adjustment">Impact of HTEs on gains from covariate adjustment</a></li>
  <li><a href="#a-hypothetical-example" id="toc-a-hypothetical-example" class="nav-link" data-scroll-target="#a-hypothetical-example">A Hypothetical Example</a></li>
  <li><a href="#application-to-geographic-atrophy-ga" id="toc-application-to-geographic-atrophy-ga" class="nav-link" data-scroll-target="#application-to-geographic-atrophy-ga">Application to Geographic Atrophy (GA)</a>
  <ul class="collapse">
  <li><a href="#prognostic-model-development" id="toc-prognostic-model-development" class="nav-link" data-scroll-target="#prognostic-model-development">Prognostic Model Development</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  <li><a href="#model-in-action" id="toc-model-in-action" class="nav-link" data-scroll-target="#model-in-action">Model in Action</a></li>
  </ul></li>
  <li><a href="#brief-recap" id="toc-brief-recap" class="nav-link" data-scroll-target="#brief-recap">Brief Recap</a></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  <li><a href="#acknowledgements" id="toc-acknowledgements" class="nav-link" data-scroll-target="#acknowledgements">Acknowledgements</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix">Appendix</a>
  <ul class="collapse">
  <li><a href="#essi-formulas-1" id="toc-essi-formulas-1" class="nav-link" data-scroll-target="#essi-formulas-1">ESSI Formulas</a></li>
  <li><a href="#huber-white-sandwich-estimator" id="toc-huber-white-sandwich-estimator" class="nav-link" data-scroll-target="#huber-white-sandwich-estimator">Huber White Sandwich Estimator</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content">





<section id="new-insights-into-an-old-method" class="level1">
<h1>New Insights into an Old Method</h1>
<center>
<p><img src="images/new_insights.png" class="img-fluid" style="width:60.0%"></p>
</center>
<p>In randomized trials, a wide range of baseline covariates are collected prior to randomization. These include demographic variables, baseline outcome assessments, biomarkers from biological samples and medical images. Baseline covariates that are prognostic for trial outcomes are often used for enrichment, stratified randomization, imbalance checking and subgroup analyses. However, it is not widely recognized that one of the most powerful uses of prognostic baseline variables in randomized clinical trials is for covariate adjustment. <em>A pre-specified primary endpoint analysis that adjusts for prognostic baseline covariates is a rigorous and virtually risk-free analysis that increases precision and power.</em></p>
<p>While the concept of covariate adjustment and its potential value for increasing trial power is decades old, historically there has been some controversy around how to ensure the validity of trial analyses without relying on model assumptions. This has been particularly challenging for non-continuous endpoints. Fortunately, there have been advancements that have resolved this controversy. Our first aim for this blog is to build on those advancements and provide concrete guidance on how to perform treatment effect estimation and inference using covariate adjustment that is both rigorous and easy-to-implement. In general, standard ANCOVA-based inference is correct <em>when the regression model is correctly specified</em>, whereas the covariate adjustment methods proposed here are asymptotically robust to model misspecification.</p>
<p>Furthermore, the pre-specified covariates that are currently used in clinical trials can be sub-optimal and strategies are needed to better determine which baseline covariates are best to use for adjustment. These baseline covariates could be a simple collection of individual variables or a single prediction from a complex machine learning model. External data, e.g.&nbsp;data from historical trials and observational studies, are an excellent resource for identifying and evaluating such candidate covariates. Accordingly, our second aim is to provide strategic guidance and relevant performance metrics for these evaluations.</p>
<p>This document will be a practical how-to guide for clinical trial statisticians on effective, trustworthy, and practical use of covariate adjustment in clinical development for maximizing power and minimizing bias. We will also discuss the value of stratified randomization compared to covariate adjustment. We focus here on continuous outcomes, but many of the points discussed apply to binary and ordinal outcomes as well.</p>
</section>
<section id="covariate-adjustment-has-gotten-hip" class="level1">
<h1>Covariate Adjustment Has Gotten Hip</h1>
<center>
<p><img src="images/collage.png" class="img-fluid" style="width:80.0%"></p>
</center>
<p>Covariate adjustment has recently gained popularity in a variety of industries. In drug development, the FDA has provided a <a href="https://www.fda.gov/regulatory-information/search-fda-guidance-documents/adjusting-covariates-randomized-clinical-trials-drugs-and-biological-products">draft guidance document</a> that strongly encourages its use. There are also several start-ups who offer covariate adjustment to optimize analysis of online experiments as a key use case for their prognostic models <span class="citation" data-cites="Netflix CUPED CUPAC MLRATE">(<a href="#ref-Netflix" role="doc-biblioref">Wong, Lewis, and Wardrop 2019</a>; <a href="#ref-CUPED" role="doc-biblioref">Deng et al. 2013</a>; <a href="#ref-CUPAC" role="doc-biblioref">Li, Tang, and Bauman 2020</a>; <a href="#ref-MLRATE" role="doc-biblioref">Guo et al. 2022</a>)</span>. Across these different applications, what statisticians and the FDA call covariate adjustment is frequently given alternative names, for example CUPED (controlled-experiment using pre-experiment data), CUPAC (Control Using Predictions As Covariates) and MLRATE (machine learning regression-adjusted treatment effect estimator). Companies like Owkin and Unlearn.AI focus on the use of covariate adjustment in randomized trials, hoping to potentially decrease trial sample size by incorporating machine learning <span class="citation" data-cites="owkin unlearn">(<a href="#ref-owkin" role="doc-biblioref">Trower, Balazard, and Patel 2020</a>; <a href="#ref-unlearn" role="doc-biblioref">Schuler et al. 2020</a>)</span>.</p>
<p>We heartily celebrate this increased enthusiasm for covariate adjustment in a variety of settings and hope the current blog post enables clinical trial statisticians to make the most effective use of baseline prognostic data.</p>
</section>
<section id="whats-the-big-deal-about-covariate-adjustment" class="level1">
<h1>What’s the Big Deal About Covariate Adjustment?</h1>
<section id="covariate-adjustment-increases-precision-and-power" class="level2">
<h2 class="anchored" data-anchor-id="covariate-adjustment-increases-precision-and-power">Covariate Adjustment Increases Precision and Power</h2>
<div id="fig-cov-adj" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cov-adj-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/increased_precision.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cov-adj-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Covariate adjustment removes explained variability
</figcaption>
</figure>
</div>
<p>Unexplained variability in the outcome data limits the precision with which you can estimate your marginal average treatment effect (ATE). If baseline covariates are prognostic, subtracting off any reasonable linear combination of those covariates from the observed outcomes, and analyzing the differences instead reduces variability of the treatment effect estimate. The marginal ATE estimate calculated using the residuals (observed-predicted) is still asymptotically unbiased, because the expected value of the predictions is equal across all treatment arms. However, the residuals have less variability than the original outcomes, leading to a more precise estimate.</p>
</section>
<section id="covariate-adjustment-removes-the-impact-of-imbalance" class="level2">
<h2 class="anchored" data-anchor-id="covariate-adjustment-removes-the-impact-of-imbalance">Covariate Adjustment Removes the Impact of Imbalance</h2>
<p>A common concern in randomized trials is that treatment groups, by chance, differ in the distributions of one or more important prognostic baseline covariates. When this occurs, one perspective is that this is not a problem since the estimates are still unbiased and type 1 error is still preserved. However, such statistical properties are unconditional on the observed covariate imbalance. When we do condition on observed imbalance, the treatment effect estimates can be biased and type 1 errors can drastically differ from their nominal levels. A simple, clear and insightful analysis of this issue was provided by Stephen Senn who has amusingly remarked “If you are at 35,000 ft, four engines are on fire and the captain has had a heart-attack can you say: ‘Why worry, on average air travel is very safe?’” <span class="citation" data-cites="Senn Senn2010">(<a href="#ref-Senn" role="doc-biblioref">S. J. Senn 1989</a>; <a href="#ref-Senn2010" role="doc-biblioref">S. Senn 2010</a>)</span>.</p>
<div id="fig-imbalence" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-imbalence-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/imbalance.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-imbalence-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Conditional type-1 error depending on imbalance
</figcaption>
</figure>
</div>
<p>How do you address concerns over imbalance? Following the analysis from Stephen Senn, the figure above shows the conditional type 1 error as a function of standardized observed imbalance for a range of assumed correlations between the baseline covariate and outcome. This shows that significance tests for imbalance (a commonly used and misguided practice) can fail to flag imbalance effects even when the conditional probability of a type 1 error well exceeds the nominal level. Therefore, significance tests do not reliably control type 1 error. Instead, Senn showed that covariate adjustment removes conditional bias from treatment effect estimates and leads to correct conditional type 1 errors of constant size.</p>
<blockquote class="blockquote">
<p>“Analysis of covariance can be recommended on two grounds: increased power and constant conditional size. The former should be sufficient to recommend the method to those who consider that the latter is irrelevant but for those who are concerned about conditional size this is an added advantage” <span class="citation" data-cites="Senn">(<a href="#ref-Senn" role="doc-biblioref">S. J. Senn 1989</a>)</span>.</p>
</blockquote>
</section>
<section id="covariate-adjustment-is-rigorous" class="level2">
<h2 class="anchored" data-anchor-id="covariate-adjustment-is-rigorous">Covariate Adjustment is Rigorous</h2>
<p>To address the historical confusion and controversy around covariate adjustment, methodological development has clearly specified the sampling framework, target estimand (i.e.&nbsp;population level parameter) and required assumptions. While we focus on continuous outcomes, the rigor of covariate adjustment also applies to non-continuous outcomes.</p>
<section id="sampling-framework" class="level3">
<h3 class="anchored" data-anchor-id="sampling-framework">Sampling framework</h3>
<p>An important but often overlooked component of specifying an estimand and estimator is the assumed sampling framework. A rigorous, adjusted analysis makes the assumed sampling framework transparent. There are several choices for the sampling framework, but we assume here the commonly used super-population framework. Let <span class="math inline">\(N\)</span> be the total sample size for a randomized trial with several treatment arms and let <span class="math inline">\(A=0,1,\ldots,a\)</span> be a random variable denoting treatment arm assignment. In the super-population framework, <span class="math inline">\(N\)</span> full data vectors are assumed to be drawn independently from some unknown, joint distribution. Patient <span class="math inline">\(i's\)</span> full data vector <span class="math inline">\((X_i, Y_i(0),Y_i(1),\ldots,Y_i(a))\)</span> contains baseline covariates <span class="math inline">\(X_i\)</span> and a potential outcome <span class="math inline">\(Y_i(k)\)</span> for each possible treatment assignment <span class="math inline">\(k=0,1,\ldots,a\)</span>. Patient <span class="math inline">\(i's\)</span> observed data vector <span class="math inline">\((A_i, X_i,
Y_i)\)</span> consists of their treatment assignment <span class="math inline">\(A_i\)</span>, baseline covariates <span class="math inline">\(X_i\)</span> and <em>observed</em> outcome <span class="math inline">\(Y_i=Y_i(0)I(A_i=0)+\ldots+Y_i(a)I(A_i=a)\)</span>. In the case of simple random sampling, <span class="math inline">\(A\)</span> and <span class="math inline">\(X\)</span> are independent.</p>
<p>It is worth noting that rigorous theory behind covariate adjustment has also been developed under the Neyman framework, which assumes a fixed, finite population where the only source of randomness is treatment assignment <span class="citation" data-cites="lin2013 Peng Ding">(<a href="#ref-lin2013" role="doc-biblioref">Lin 2013</a>; <a href="#ref-Peng" role="doc-biblioref">Ding, Li, and Miratrix 2017</a>, <a href="#ref-Ding" role="doc-biblioref">2019</a>)</span>.</p>
</section>
<section id="target-estimand" class="level3">
<h3 class="anchored" data-anchor-id="target-estimand">Target Estimand</h3>
<p>Confusion often arises since there are various kinds of covariate adjusted analyses targeting different estimands. For example, covariate adjustment is often used in the estimation of conditional average treatment effects, which are contrasts in treatment arm means that are conditional on baseline covariates. A conditional treatment effect is not a single value, but rather a function of baseline covariates, unless one makes the assumption that the conditional treatment effect is constant.</p>
</section>
</section>
</section>
<section id="conditional-average-treatment-effect" class="level1 callout-definition">
<h1>Conditional Average Treatment Effect</h1>
<p>A contrast (difference, ratio, etc.) between treatment arm means conditional on baseline covariates. Example of a conditional ATE:</p>
<p><span id="eq-cond-avg-trt-eff"><span class="math display">\[
E(Y_{active}|X) - E(Y_{control}|X)
\tag{1}\]</span></span></p>
</section>
<p>However, in this document, we do not use covariate adjustment for that purpose. Instead, we us <strong>covariate adjustment as a tool to more efficiently estimate the marginal ATE</strong>:</p>
<section id="marginal-average-treatment-effect" class="level1 callout-definition">
<h1>Marginal Average Treatment Effect</h1>
<p>A contrast (difference, ratio, etc.) between marginal treatment arm means. Example of a marginal ATE:</p>
<p><span id="eq-marg-avg-trt-eff"><span class="math display">\[
E(Y_{active}) - E(Y_{control})
\tag{2}\]</span></span></p>
</section>
<p>Some researchers suggest focusing on estimating conditional treatment effects and using covariate adjustment for that purpose. While estimating conditional and individualized treatment effects is an important research objective, a discussion about how to do so is beyond the scope of this work. Here, we assume estimating the marginal ATE is the primary objective of a label-enabling clinical trial.</p>
<p>We note that the terms ‘marginal’ and ‘conditional’ are often conflated with ‘unadjusted’ and ‘adjusted’, however these terms are not synonymous <span class="citation" data-cites="Daniel">(<a href="#ref-Daniel" role="doc-biblioref">Daniel, Zhang, and Farewell 2020</a>)</span>:</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Marginal vs.&nbsp;Conditional
</div>
</div>
<div class="callout-body-container callout-body">
<p>Used to distinguish different kinds of estimands.</p>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Unadjusted vs.&nbsp;Adjusted
</div>
</div>
<div class="callout-body-container callout-body">
<p>Used to distinguish different kinds of estimators.</p>
</div>
</div>
<section id="assumptions" class="level3">
<h3 class="anchored" data-anchor-id="assumptions">Assumptions</h3>
<p>Historically, researchers have debated how to safely estimate marginal ATEs using covariate adjustment. Some have been hesitant to use covariate adjustment over concerns around model misspecification, bias and possible losses in precision <span class="citation" data-cites="Freedman2008a Freedman2008b">(<a href="#ref-Freedman2008a" role="doc-biblioref">D. A. Freedman 2008a</a>; <a href="#ref-Freedman2008b" role="doc-biblioref">David A. Freedman 2008b</a>)</span>, while others have enthusiastically used covariate adjustment without a second thought.</p>
<p>Academic research ultimately settled that debate <span class="citation" data-cites="Tsiatis2001 Tsiatis2008 Rosenblum2009 TMLE2010">(<a href="#ref-Tsiatis2001" role="doc-biblioref">Yang and Tsiatis 2001</a>; <a href="#ref-Tsiatis2008" role="doc-biblioref">Tsiatis et al. 2008</a>; <a href="#ref-Rosenblum2009" role="doc-biblioref">Rosenblum and Laan 2009</a>; <a href="#ref-TMLE2010" role="doc-biblioref">Rosenblum and Laan 2010</a>)</span>. A number of approaches to covariate adjustment have been characterized that result in consistent, asymptotically normal estimators of marginal ATEs even if the regression model is misspecified. This means that even if important interactions or non-linearities are excluded, prognostic factors are missed, or factors are included that have no prognostic value, covariate adjustment consistently estimates the marginal ATE.</p>
<p>The covariate adjustment approach we use is called a <a href="#standardized-regression-estimator">standardized regression estimator</a>, which uses predictions from a working regression model to obtain marginal ATEs. The model is referred to as a working regression model because it does not have to represent the true data generating distribution. Therefore, we refer to covariate adjustment as a <strong>model-assisted, not model-dependent, analysis</strong>. Examples of working regression models include logistic, OLS, and Poisson regression models.</p>
<p>For the standardized regression estimator, it is sufficient to assume that treatment is assigned via simple random sampling, that the full data vectors are i.i.d. and that all variables are bounded <span class="citation" data-cites="Rosenblum2009 TMLE2010">(<a href="#ref-Rosenblum2009" role="doc-biblioref">Rosenblum and Laan 2009</a>; <a href="#ref-TMLE2010" role="doc-biblioref">Rosenblum and Laan 2010</a>)</span>. Extensions to other sampling frameworks have also been explored <span class="citation" data-cites="Wang2021">(<a href="#ref-Wang2021" role="doc-biblioref">Wang et al. 2021</a>)</span>.</p>
</section>
<section id="covariate-adjustment-is-encouraged-by-the-fda" class="level2">
<h2 class="anchored" data-anchor-id="covariate-adjustment-is-encouraged-by-the-fda">Covariate Adjustment is Encouraged by the FDA</h2>
<p>In a 2021 <a href="https://www.fda.gov/regulatory-information/search-fda-guidance-documents/adjusting-covariates-randomized-clinical-trials-drugs-and-biological-products">draft guidance</a>, pre-specified covariate adjustment was fully supported and encouraged by the FDA as a primary endpoint analysis in clinical trials:</p>
<blockquote class="blockquote">
<p>“Although an unadjusted analysis is acceptable for the primary analysis, adjustment for baseline covariates will generally reduce the variability of estimation of treatment effects and thus lead to narrower confidence intervals and more powerful hypothesis testing” <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>.</p>
</blockquote>
<p>Importantly, the guidance states that “[s]ponsors should prospectively specify the covariates and the mathematical form of the covariate adjusted estimator” <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>. We interpret this to mean that the exact working regression model form should be pre-specified before working with the trial data. Independent data from observational cohorts or previous trials can be used to select or develop covariates. These covariates may be existing baseline variables or even the output from a multivariable prediction model which takes as input other baseline covariates.</p>
<p>In contrast to pre-specifying an ANCOVA regression model, there is active research into applying machine learning directly to the unblinded trial data under investigation (as opposed to historical data) <span class="citation" data-cites="covid2 covid Tsiatis2008 Tian2012">(<a href="#ref-covid2" role="doc-biblioref">Williams, Rosenblum, and Diaz 2021</a>; <a href="#ref-covid" role="doc-biblioref">Benkeser et al. 2020</a>; <a href="#ref-Tsiatis2008" role="doc-biblioref">Tsiatis et al. 2008</a>; <a href="#ref-Tian2012" role="doc-biblioref">Tian et al. 2012</a>)</span> as part of the covariate adjusted analysis. While the role machine learning procedures can play in covariate adjustment is an interesting and important area of research, we do not yet recommend using them as part of the primary endpoint analysis of a clinical trial due to several considerations. First, the finite sample properties of the estimators are not understood. Second, these methods add complexity and burden to the study yet are unlikely to provide meaningful benefits.</p>
<p>It may not be appreciated that a lot of things need to happen between the unblinding of a randomized clinical trial and the reporting of topline results. Outputs and code are quality controlled, pre-specified sensitivity analyses are performed, and internal reviews are conducted with key decision makers and stakeholders. All of these activities need to occur in a tight time frame for practical considerations such as appropriate control of insider information.</p>
<p>In addition, during the planning phase, many choices have to be made when applying machine learning directly to the trial data such as which covariates to present to the machine learning procedure, the method of cross validation to use, which machines to include if using ensemble learning and how to perform standard error estimation. These choices, many of which are based on simulation studies, add to the difficulty of finalizing the statistical analysis plan.</p>
<p>In summary, applying machine learning to the trial data as part of the primary endpoint analysis adds considerable burden to the study team. <em>This added burden would only be worthwhile if the expected precision gains were meaningful</em>. However, in our experience, complex black-box models rarely outperform simple linear models when working with tabular data and do not support this additional complexity and time.</p>
</section>
<section id="how-to-perform-covariate-adjustment" class="level1 page-columns page-full">
<h1>How to Perform Covariate Adjustment</h1>
<section id="standardized-regression-estimator" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="standardized-regression-estimator">Standardized Regression Estimator</h2>
<p>Table 1 provides instructions and code for estimating the Treatment Arm Means (TAMs) and the marginal ATE (in this case, the difference in means) using covariate adjustment. The example code shows how to estimate these quantities for trials with two treatment arms, but easily generalizes to more than two arms. This estimator is often referred to as the <strong>standardized regression estimator</strong>, and is also an example of a targeted maximum likelihood estimator. There are other consistent estimators of the marginal ATE that use covariate adjustment (see <span class="citation" data-cites="Colantuoni2015">Colantuoni and Rosenblum (<a href="#ref-Colantuoni2015" role="doc-biblioref">2015</a>)</span> for examples). However, we suggest using the standardized regression estimator because it is easy to use, “statistically reliable” <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>, has comparable power and can also be used to estimate TAMS and marginal ATEs for non-continuous outcomes.</p>
<p>In the notation below, <span class="math inline">\(A\)</span> is a two-level factor indicating treatment assignment, <span class="math inline">\(Y\)</span> is the observed outcome, <span class="math inline">\(X\)</span> is a baseline covariate, <span class="math inline">\(E[Y|a,
X_i] = \mu(a,X_i)\)</span> is the expected value of <span class="math inline">\(Y\)</span> for patient <span class="math inline">\(i\)</span> given their treatment assignment <span class="math inline">\(A_i=a\)</span> and baseline covariate <span class="math inline">\(X_i\)</span>, <span class="math inline">\(\hat{\mu}(a,X_i)\)</span> is the predicted outcome for patient <span class="math inline">\(i\)</span> from the working regression model and an estimate of <span class="math inline">\(\mu(a,X_i)\)</span>, and <span class="math inline">\(N\)</span> is the total trial sample size. The standardized regression estimator for the mean outcome in treatment arm <span class="math inline">\(A=a\)</span> is:</p>
<p><span id="eq-std-reg-mean"><span class="math display">\[
\widehat{TAM_a}=\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(a,X_i)
\tag{3}\]</span></span></p>
<p>The standardized regression estimator for the difference in means for treatment arms <span class="math inline">\(A=0\)</span> and <span class="math inline">\(A=1\)</span> is:</p>
<p><span id="eq-std-reg-diff"><span class="math display">\[
\widehat{ATE}=\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(1,X_i)-\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(0,X_i)
\tag{4}\]</span></span></p>
<p>Other contrasts of the marginal means (TAMs) may be similarily estimated.</p>
<div class="column-page-inset">
<div id="tbl-estimation" class="quarto-float anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-estimation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Standardized regression estimator for TAMs and marginal ATE
</figcaption>
<div aria-describedby="tbl-estimation-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="table">
<colgroup>
<col style="width: 5%">
<col style="width: 27%">
<col style="width: 64%">
<col style="width: 0%">
</colgroup>
<thead>
<tr class="header">
<th>Step</th>
<th>Estimation Instructions</th>
<th colspan="2">Example R Code</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Fit a working regression model using data from all treatment arms, regressing the outcome on treatment and prognostic baseline covariates</td>
<td><div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Additive Working Regression Models</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>mod.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y <span class="sc">~</span> A <span class="sc">+</span> X, <span class="at">data =</span> d)  <span class="co"># continuous</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>mod.fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(Y <span class="sc">~</span> A <span class="sc">+</span> X, <span class="at">data =</span> d, <span class="at">family =</span> binomial)  <span class="co"># binary</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Working Regression Models with Interaction</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>mod.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y <span class="sc">~</span> A <span class="sc">*</span> X, <span class="at">data =</span> d)  <span class="co"># continuous</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>mod.fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(Y <span class="sc">~</span> A <span class="sc">*</span> X, <span class="at">data =</span> d, <span class="at">family =</span> binomial)  <span class="co"># binary</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>For each subject, use the model from step 1 and the subject’s baseline covariates to compute their predicted outcome under EACH treatment of interest (regardless of what their assigned treatment was)</td>
<td><div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Continuous outcome</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>pred0 <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod.fit, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">A=</span><span class="dv">0</span>, <span class="at">X=</span>d<span class="sc">$</span>X))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>pred1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod.fit, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">A=</span><span class="dv">1</span>, <span class="at">X=</span>d<span class="sc">$</span>X))</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Binary outcome</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>pred0 <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod.fit, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">A=</span><span class="dv">0</span>, <span class="at">X=</span>d<span class="sc">$</span>X), <span class="at">type=</span>‘response’)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>pred1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod.fit, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">A=</span><span class="dv">1</span>, <span class="at">X=</span>d<span class="sc">$</span>X, <span class="at">type=</span>‘response’)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>Take the average of the predicted outcomes in each treatment group to get estimates of the Treatment Arm Means (TAMs)</td>
<td><div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean of predictions in treatment arm A=0</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>TAM0 <span class="ot">&lt;-</span> <span class="fu">mean</span>(pred0)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean of predictions in treatment arm A=1</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>TAM1 <span class="ot">&lt;-</span> <span class="fu">mean</span>(pred1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>Compute the desired contrast of the TAMs to get an estimate of the marginal ATE</td>
<td><div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate the desired contrast, for example, the</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># difference between treatment arm A=1 and A=0</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>ATE <span class="ot">&lt;-</span> TAM1 <span class="sc">-</span> TAM0</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
</div>
<p>In the example code in Table 1, <span class="math inline">\(X\)</span> is a single covariate, but <span class="math inline">\(X\)</span> can also be a matrix containing a set of individual covariates, a prediction from an independent prognostic model, or a combination thereof. When the working regression model is an additive OLS regression, the standardized regression estimate of the difference in means marginal ATE equals the estimated coefficient for <span class="math inline">\(A\)</span>. This is also true for an interaction model when the covariates are centered. Note that, unlike OLS regression, in generalized linear models the estimated coefficient for treatment does not translate into a marginal ATE.</p>
<p>Theoretically, it is possible to lose precision with covariate adjustment when using an additive model compared to an unadjusted analysis. In contrast, a standardized regression estimator that uses an interaction working regression model is asymptotically guaranteed to be at least as precise as an unadjusted estimator or adjustment with an additive model. However, it is only under unrealistic conditions that covariate adjustment with an additive model will be less precise than an unadjusted estimator. What is more, if randomization is 1:1 or the covariances between covariates and outcome are equal across treatment arms, then covariate adjustment with an additive model will be as efficient as with an interaction model. In our experience, interaction models do not lead to considerable gains in precision over additive models. Likewise, Tsiatis et al. (2001) performed simulations comparing precision when using an additive vs.&nbsp;an interaction model and “found the loss of efficiency to be trivial in all cases” <span class="citation" data-cites="Tsiatis2001">(<a href="#ref-Tsiatis2001" role="doc-biblioref">Yang and Tsiatis 2001</a>)</span>. Therefore, <strong>additive models are a good default choice</strong> especially if your trial sample size is too small to support treatment-by-covariate interactions (see <a href="#determining-your-model-budget">model budget section</a>).</p>
<p>In case of missing values in baseline covariates, the FDA draft guidance suggests that “covariate adjustment is generally robust to the handling of subjects with missing baseline covariates. Missing baseline covariate values can be singly or multiply imputed, or missingness indicators (Groenwold et al.&nbsp;2012) can be added to the model used for covariate adjustment. Sponsors should not perform imputation separately for different treatment groups, and sponsors should ensure that imputed baseline values are not dependent on any post-baseline variables, including the outcome” <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>. A baseline covariate that has too many missing values should not be used for adjustment, but if the number of missing values is small, we recommend simple, single imputation as opposed to using indicator variables in order to spend the model budget wisely.</p>
</section>
<section id="inference-for-tams-and-ates" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="inference-for-tams-and-ates">Inference for TAMs and ATEs</h2>
<p>We suggest using robust standard error estimators from the targeted maximum likelihood estimation (TMLE) literature because they are robust to model misspecification, are fast and easy to implement with a few lines of code, are valid under any randomization ratio and data generating distribution, provide standard error estimates for both ATEs and treatment arm means, are aligned with the superpopulation framework and are deterministic.</p>
<p>Let <span class="math inline">\(I(A_i=a)\)</span> be an indicator of patient <span class="math inline">\(i\)</span> receiving treatment <span class="math inline">\(A=a\)</span> and <span class="math inline">\(p(A=a)\)</span> be the probability of being assigned to treatment arm <span class="math inline">\(A=a\)</span>. The TMLE <strong>standard error estimate for the mean of treatment arm <span class="math inline">\(A=a\)</span></strong> is:</p>
<p><span id="eq-std-err-est-mean"><span class="math display">\[
\sqrt{\frac{\frac{1}{N}\sum_{i=1}^{N}\Bigg(\frac{I(A_i=a)(Y_i-\hat{\mu}(a,X_i))}{p(A=a)}+\hat{\mu}(a,X_i)-\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(a,X_i)\Bigg)^2}{N}}
\tag{5}\]</span></span></p>
<p>The standard error estimate for the <strong>difference in Means for Treatment arms <span class="math inline">\(A=1\)</span> and <span class="math inline">\(A=0\)</span></strong> is:</p>
<div class="column-body-outset">
<p><span id="eq-std-err-est-diff"><span class="math display">\[
\sqrt{\frac{\frac{1}{N}\sum_{i=1}^{N}\Bigg(\frac{I(A_i=1)(Y_i-\hat{\mu}(1,X_i))}{p(A=1)}+\hat{\mu}(1,X_i)-\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(1,X_i) - \Big(\frac{I(A_i=0)(Y_i-\hat{\mu}(0,X_i))}{p(A=0)}+\hat{\mu}(0,X_i)-\frac{1}{N}\sum_{i=1}^{N}\hat{\mu}(0,X_i)\Big) \Bigg)^2}{N}}
\tag{6}\]</span></span></p>
</div>
<p>The code for the above SE estimators is shown below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## SE estimate for treatment arm mean A=0</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="fu">mean</span>(((A <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">*</span> (Y <span class="sc">-</span> pred0) <span class="sc">/</span> <span class="fu">mean</span>(A <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">+</span> pred0 <span class="sc">-</span> <span class="fu">mean</span>(pred0)) <span class="sc">^</span> <span class="dv">2</span>) <span class="sc">/</span> <span class="fu">nrow</span>(d))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="do">## SE estimate for treatment arm mean A=1</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="fu">mean</span>(((A <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">*</span> (Y <span class="sc">-</span> pred1) <span class="sc">/</span> <span class="fu">mean</span>(A <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">+</span> pred1 <span class="sc">-</span> <span class="fu">mean</span>(pred1)) <span class="sc">^</span> <span class="dv">2</span>) <span class="sc">/</span> <span class="fu">nrow</span>(d))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="do">## SE estimate for the marginal ATE, the difference between treatment arm mean A=1 and A=0</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="fu">mean</span>(((A <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">*</span> (Y <span class="sc">-</span> pred1) <span class="sc">/</span> <span class="fu">mean</span>(A <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">+</span> pred1 <span class="sc">-</span> <span class="fu">mean</span>(pred1) <span class="sc">-</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>        ((A <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">*</span> (Y <span class="sc">-</span> pred0) <span class="sc">/</span> <span class="fu">mean</span>(A <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">+</span> pred0 <span class="sc">-</span> <span class="fu">mean</span>(pred0))) <span class="sc">^</span> <span class="dv">2</span>) <span class="sc">/</span> <span class="fu">nrow</span>(d))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Using the estimated standard errors and the asymptotic normality of the standardized regression estimator, one can then easily obtain <span class="math inline">\(p\)</span>-values and <span class="math inline">\(95\%\)</span> confidence intervals for both the treatment arm means and the marginal ATE.</p>
<p>We note that the above standard error estimates assume observations are i.i.d. (i.e.&nbsp;randomization is not stratified), and we address this assumption in a <a href="#accounting-for-stratified-randomization">later section</a>. For standard error estimates corresponding to marginal ATEs using different contrasts (e.g.&nbsp;ratio instead of difference) see Rosenblum and van der Laan (2010).</p>
<p>Alternative standard error estimators include the Huber-White estimator and re-sampling based methods like the non-parametric bootstrap. In their recent guidance, the FDA recommends using the Huber-White robust “sandwich” estimator instead of nominal standard error estimates from statistical software <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>. However, if using a working regression model with treatment-by-covariate interactions, the expected sandwich standard error estimates may be too small when treatment effects are heterogeneous (<a href="#appendix">see Appendix</a> and <span class="citation" data-cites="imbens">Imbens and Wooldridge (<a href="#ref-imbens" role="doc-biblioref">2009</a>)</span> , <span class="citation" data-cites="lin2013">Lin (<a href="#ref-lin2013" role="doc-biblioref">2013</a>)</span>). Bootstrap methods are a reasonable alternative, but we prefer TMLE estimators since they are easier to implement and less computationally intensive. Furthermore, bootstrap methods lead to confidence intervals and p-values that are not deterministic. An intriguing alternative to the bootstrap that is deterministic is the jackknife <span class="citation" data-cites="jackknife">(<a href="#ref-jackknife" role="doc-biblioref">Wolbers et al. 2022</a>)</span>.</p>
</section>
<section id="determining-your-model-budget" class="level2">
<h2 class="anchored" data-anchor-id="determining-your-model-budget">Determining your Model Budget</h2>
<p>Is there such a thing as adjusting for too many prognostic covariates? Yes! The marvelous properties of covariate adjustment discussed thus far (consistency, robustness to model misspecification, precision gains, etc.) are <strong>large sample properties</strong>. Thus, adjusting for too many covariates in your working model relative to your sample size may invalidate the analysis <span class="citation" data-cites="ML Colantuoni2015 Steingrimsson2016">(<a href="#ref-ML" role="doc-biblioref">Rosenblum 2020</a>; <a href="#ref-Colantuoni2015" role="doc-biblioref">Colantuoni and Rosenblum 2015</a>; <a href="#ref-Steingrimsson2016" role="doc-biblioref">Rosenblum and Steingrimsson 2016</a>)</span>, and the FDA advises against adjusting for too many covariates:</p>
<blockquote class="blockquote">
<p>“The statistical properties of covariate adjustment are best understood when the number of covariates adjusted for in the study is small relative to the sample size.”<span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span></p>
</blockquote>
<p>Explicit guidance on how many covariates to adjust for is not offered, so we borrow from prognostic modeling practices <span class="citation" data-cites="rms">(<a href="#ref-rms" role="doc-biblioref">Harrell 2011</a>)</span>.</p>
</section>
</section>
<section id="model-budget" class="level1 callout-definition">
<h1>Model Budget</h1>
<p>The number of allowable terms in the working regression model, excluding the overall intercept. Suggestions for a model budget based on classic events per variable considerations are <span class="citation" data-cites="rms">(<a href="#ref-rms" role="doc-biblioref">Harrell 2011</a>)</span>:</p>
<table class="table">
<colgroup>
<col style="width: 46%">
<col style="width: 7%">
<col style="width: 46%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">terms (% of total sample size)</th>
<th style="text-align: center;"></th>
<th style="text-align: left;">guidance</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;"><span class="math inline">\(\leq 5\%\)</span></td>
<td style="text-align: center;"><i class="bi bi-check-circle-fill text-success"></i></td>
<td style="text-align: left;">likely a safe choice</td>
</tr>
<tr class="even">
<td style="text-align: right;"><span class="math inline">\(\approx7.5\%\)</span></td>
<td style="text-align: center;"><i class="bi bi-exclamation-triangle-fill text-warning"></i></td>
<td style="text-align: left;">probably reasonable</td>
</tr>
<tr class="odd">
<td style="text-align: right;"><span class="math inline">\(\gt 10\%\)</span></td>
<td style="text-align: center;"><i class="bi bi-question-octagon-fill text-danger"></i></td>
<td style="text-align: left;">potentially unsafe</td>
</tr>
</tbody>
</table>
</section>
<p>Pre-specifying a set of covariates and a working regression model form that satisfies the model budget maintains the rigor and integrity of the statistical analysis. We note that this is a rough guidance and should be considered a useful starting point. Recently, model budget calculations for general prognostic modeling have been refined and in future work it would be helpful to adapt such an approach to covariate adjustment (see <span class="citation" data-cites="Riley2019">R. D. Riley et al. (<a href="#ref-Riley2019" role="doc-biblioref">2019</a>)</span> and <span class="citation" data-cites="Riley2020">R. Riley et al. (<a href="#ref-Riley2020" role="doc-biblioref">2020</a>)</span>).</p>
<section id="spending-your-model-budget-wisely" class="level2">
<h2 class="anchored" data-anchor-id="spending-your-model-budget-wisely">Spending your Model Budget Wisely</h2>
<center>
<p><img src="images/model_budget.png" class="img-fluid" style="width:60.0%"></p>
</center>
<p>The model budget tells you how complex the working regression model can be while ensuring treatment effect estimates are unbiased, confidence intervals have good coverage, and type 1 error rates for hypothesis testing do not exceed nominal levels. However, there are many options when pre-specifying the working regression model and it is easy to spend beyond one’s model budget, so strategic choices will need to be made prior to looking at the data.</p>
<p>We suggest determining the model form and spending the model budget according to the following four principles:</p>
<ol type="1">
<li><p><strong>Leverage external/historical data</strong><br>
Historically, pre-specification of a regression model was guided by ad hoc analyses reported in the literature, input from subject matter experts, and precedent set by previous trials. While these continue to be important considerations, big data collection and curation and the introduction of machine learning methods to drug development significantly improve the process of pre-specifying the regression model form.</p></li>
<li><p><strong>Prioritize maximally prognostic covariates</strong><br>
The list of candidate prognostic covariates may be too large according to the model budget, in which case a subset of prognostic covariates will need to be selected. We use external/historical data to guide that selection (see upcoming section on <a href="#performance-metrics">Performance Metrics</a>).</p></li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Pro Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>If the model budget is tight and several covariates are expected to meaningfully add prognostic value, consider combining them into a single score and using that in the model instead of the individual covariates.</p>
</div>
</div>
<div id="fig-prog-mod" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-prog-mod-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/prognostic_modeling.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-prog-mod-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Leverage independent data to find the most promising prognostic factors/models for covariate adjustment
</figcaption>
</figure>
</div>
<ol start="3" type="1">
<li><p><strong>Avoid Dichotomania!</strong><br>
Usually the relationship between a continuous variable and outcome varies gradually and abrupt dichotomization loses prognostic information <span class="citation" data-cites="Ge2011">Stephen J. Senn (<a href="#ref-Dichotomania" role="doc-biblioref">2005</a>)</span>. We recommend that, unless there is evidence to the contrary, prognostic covariates that have been discretized be replaced with their underlying continuous forms. Note that, in particular, this will impact how stratification factors enter into the model.</p></li>
<li><p><strong>By default, do not include interactions between covariates</strong><br>
There is a large body of evidence showing that for tabular data, purely additive models are very rarely, if ever, meaningfully improved by including interactions <span class="citation" data-cites="Rudin">(<a href="#ref-Rudin" role="doc-biblioref">Rudin 2019</a>)</span>. This has also been our experience with many prognostic modeling projects. We do not recommend including interactions between prognostic covariates in the model since these are usually wasteful expenses in the model budget, unless evidence to the contrary is available.</p></li>
<li><p><strong>By default, do not include interactions between covariates and treatment arm</strong><br>
<a href="#standardized-regression-estimator">As discussed above</a>, such interactions rarely provide meaningfully improved precision and power under realistic assumptions.</p></li>
</ol>
<p>Use the above five principles to pre-specify a working regression model and to spend your model budget wisely. Then, use this model in the <a href="#standardized-regression-estimator">previously described standardized estimator</a> to estimate the treatment arm means and marginal ATEs and to <a href="#inference-for-tams-and-ates">perform inference</a>.</p>
</section>
<section id="accounting-for-stratified-randomization" class="level2">
<h2 class="anchored" data-anchor-id="accounting-for-stratified-randomization">Accounting for Stratified Randomization</h2>
<p>The <a href="#standardized-regression-estimator">estimation</a> and <a href="#inference-for-tams-and-ates">inference</a> procedures described in the previous sections assume <em>simple randomization</em>, where each patient’s treatment assignment is an independent and identically distributed Bernoulli random variable for a trial with two treatment arms and a categorical random variable for a trial with more than two treatment arms. However, randomization in clinical trials is usually stratified by selected baseline covariates, for example, using permuted block randomization. In this case, the FDA recommends that the standard error estimation <em>“account for the stratified randomization (Bugni et al.&nbsp;2018) with or without strata variables in an adjustment model. Otherwise, the standard error is likely to be overestimated and interval estimation and hypothesis testing can become unduly conservative”</em> <span class="citation" data-cites="FDA">(<a href="#ref-FDA" role="doc-biblioref">FDA 2021</a>)</span>.</p>
<p><strong>However, we recommend that it is NOT necessary to modify estimation and inference to account for stratified randomization.</strong> Instead, we propose that the previously described strategy for <a href="#spending-your-model-budget-wisely">spending the model budget wisely</a> be used and applied to all prognostic candidates, making no distinction between stratification factors and other baseline covariates. Then use covariate adjustment as described in the previous sections for estimation and inference of the marginal ATEs, <em>as if treatment were allocated using simple random sampling</em>.</p>
<p>Note that, as a consequence:</p>
<ol type="1">
<li>By default, if stratification variables do enter into the working regression model, they will do so as additive terms.</li>
<li>Dichotomized stratification factors will be replaced by their underlying continuous variables.</li>
<li>Stratification variables not expected to be correlated with outcome in any treatment group may be omitted entirely.</li>
</ol>
<p>Such an approach still maintains type 1 error and leads to consistent estimators for marginal ATEs. At worst, inference is conservative (p-values may be too large and confidence intervals too wide) <span class="citation" data-cites="Wang2021">(<a href="#ref-Wang2021" role="doc-biblioref">Wang et al. 2021</a>)</span>. However, <a href="#spending-your-model-budget-wisely">if the model budget is spent wisely</a> to obtain a sensible working regression model, conservatism will be negligible.</p>
</section>
<section id="performance-metrics" class="level1">
<h1>Performance Metrics</h1>
<center>
<p><img src="images/performance.png" class="img-fluid" style="width:50.0%"></p>
</center>
<section id="intuitive-metrics-for-quantifying-the-expected-benefit-of-adjustment" class="level2">
<h2 class="anchored" data-anchor-id="intuitive-metrics-for-quantifying-the-expected-benefit-of-adjustment">Intuitive Metrics for Quantifying the Expected Benefit of Adjustment</h2>
<p>When analyzing external/historical data in preparation for pre-specifying a covariate-adjusted primary endpoint analysis, it’s important to understand the following:</p>
<ol type="1">
<li>Which covariates to adjust for</li>
<li>The expected precision gains from adjustment</li>
</ol>
<p>Below, we provide performance metrics that allow study statisticians to easily select which covariates to adjust for, to understand the impact of heterogeneous treatment effects (HTEs) on precision gains and to estimate the expected benefit from adjustment <strong>without having to perform simulations</strong>.</p>
<p>A great way to derive a single, high-quality covariate is to use external data to construct a prognostic model that combines multiple covariates into a single prognostic score. Thus, we focus here on how to evaluate a single prognostic covariate. However, if the model budget allows and the prognostic model is originally linear in form, the individual covariates can be expanded out in the working regression model and re-fit, usually with no loss in performance.</p>
<p>For practical use in a trial, prognostic models should achieve <strong>maximum performance with minimal complexity.</strong> Complexity is determined by the:</p>
<ol type="1">
<li>Cost, incovenience, and invasiveness of covariates.</li>
<li>Ease of deployment and implementation of the prognostic model.</li>
<li>Algorithmic complexity (e.g.&nbsp;linear models vs deep learners).</li>
</ol>
<p>We find that linear prognostic models are hard to beat when working with tabular data. Deep learning models are more likely to provide a boost in performance over linear models when working with imaging data. Deep learners are natural candidates when working with non-tabular data, such as raw image files. Later we will show an example where considerable gains are achieved using a deep learner trained to predict progression in an ophthalmologic disease from raw images.</p>
<p>An intuitive metric for the expected benefit of a prognostic score is the amount by which it effectively increases the study’s sample size via the gain in precision, i.e.&nbsp;the <strong>Effective Sample Size Increase (ESSI)</strong>. For example, suppose adjusting for a particular covariate is associated with an estimated ESSI of <span class="math inline">\(40\%\)</span>. That means, asymptotically, the following two analyses have the same power:</p>
<ul>
<li>A covariate-adjusted analysis with sample size <span class="math inline">\(N\)</span></li>
<li>An unadjusted analysis but with <span class="math inline">\(40\%\)</span> more patients, <span class="math inline">\(1.4 * N\)</span></li>
</ul>
<p>ESSI estimates can be used to compare sets of covariates or prognostic models and make transparent the expected benefits of adjustment.</p>
</section>
<section id="essi-formulas" class="level2">
<h2 class="anchored" data-anchor-id="essi-formulas">ESSI Formulas</h2>
<p>For simplicity, we assume 1:1 randomization and equal, marginal variances across the treatment arms. We also provide <a href="#appendix">ESSI formulas</a> that relax these assumptions.</p>
<p>It turns out that the <strong>ESSI from covariate adjustment for continuous outcomes</strong> can be easily estimated since it only depends on two parameters:</p>
</section>
</section>
<section id="general-essi" class="level1 callout-definition">
<h1>General ESSI</h1>
<p><span id="eq-essi-gen"><span class="math display">\[
General \space ESSI=\Bigg(\frac{1}{1 - \big(\frac{r_{control} \space + \space r_{active}}{2}\big)^2}-1\Bigg) * 100\%
\tag{7}\]</span></span></p>
<p><span class="math display">\[
\begin{align*}
  \text{where} \\
  r_{active} &amp;= \text{Correlation between the outcome and covariate in the active,} \\
  r_{control} &amp;= \text{Correlation between the outcome and covariate in the control arm}
\end{align*}
\]</span></p>
</section>
<p>It makes intuitive sense that the ESSI depends on the squared average of the correlations in the two treatment arms, increasing as the correlations increase. The correlation between the covariate and the outcome among untreated patients (<span class="math inline">\(r_{control}\)</span>) can be estimated using historical trial data, observational cohorts, etc.</p>
<p>This leaves <span class="math inline">\(r_{active}\)</span> as the only quantity left to estimate. To help estimate <span class="math inline">\(r_{active}\)</span>, note that the correlation between the outcome and the covariate in the treatment arm depends on the “treatment effect scenario.” Two plausible treatment effect scenarios are discussed in the following sections.</p>
<section id="constant-absolute-treatment-effect" class="level3">
<h3 class="anchored" data-anchor-id="constant-absolute-treatment-effect">Constant absolute treatment effect</h3>
<p>One commonly assumed treatment effect scenario is that the <em>absolute</em> treatment effect is constant across the baseline covariate:</p>
</section>
<section id="constant-absolute-treatment-effect-1" class="level1 callout-definition">
<h1>Constant absolute treatment effect:</h1>
<p><span class="math inline">\(E(Y_{active}|X)=E(Y_{control}|X)-\delta\)</span>, for some constant <span class="math inline">\(\delta\)</span>. With a constant absolute treatment effect, <span class="math inline">\(r_{control}=r_{active}\)</span>. Plugging this into the general ESSI formula we get</p>
<p><span id="eq-essi-const-abs-delta"><span class="math display">\[
ESSI \space Assuming \space Constant \space Absolute \space \delta =\Bigg(\frac{1}{1-r^2_{control}}-1\Bigg) * 100\%
\tag{8}\]</span></span></p>
</section>
<div class="cell">
<div class="cell-output-display">
<div id="fig-constant-ate" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-constant-ate-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="index_files/figure-html/fig-constant-ate-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-constant-ate-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Constant Absolute Treatment Effect
</figcaption>
</figure>
</div>
</div>
</div>
<section id="constant-proportional-treatment-effect" class="level3">
<h3 class="anchored" data-anchor-id="constant-proportional-treatment-effect">Constant proportional treatment effect</h3>
<p>A second plausible treatment effect scenario is that the <em>relative</em> treatment effect is constant across the baseline covariate:</p>
</section>
<section id="constant-proportional-treatment-effect-1" class="level1 callout-definition">
<h1>Constant proportional treatment effect</h1>
<p><span class="math inline">\(E(Y_{active}|X)=(1-\delta)E(Y_{control}|X)\)</span>, for some constant <span class="math inline">\(\delta\)</span>. With a constant proportional treatment effect, <span class="math inline">\(r_{active}=(1-\delta)r_{control}\)</span>. Plugging this into the general ESSI formula we get</p>
<p><span id="eq-essi-const-prop-delta"><span class="math display">\[
ESSI\space Assuming \space Constant \space Proportional \space \delta=\Bigg(\frac{1}{1-r^2_{control}\big(1-\frac{\delta}{2}\big)^2}-1\Bigg) * 100\%
\tag{9}\]</span></span></p>
</section>
<div class="cell">
<div class="cell-output-display">
<div id="fig-constant-pte" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-constant-pte-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="index_files/figure-html/fig-constant-pte-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-constant-pte-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Constant Proportional Treatment Effect
</figcaption>
</figure>
</div>
</div>
</div>
<section id="impact-of-htes-on-gains-from-covariate-adjustment" class="level2">
<h2 class="anchored" data-anchor-id="impact-of-htes-on-gains-from-covariate-adjustment">Impact of HTEs on gains from covariate adjustment</h2>
<p>A constant proportional treatment effect is an example of HTEs. A treatment effect is heterogeneous if the magnitude of the treatment effect (on the absolute scale) depends on the value of the baseline covariate. <strong>HTEs decrease the precision gains from adjustment</strong>. For example, notice how, in the constant proportional treatment effect formula, the ESSI decreases as <span class="math inline">\(\delta\)</span> increases (i.e.&nbsp;as heterogeneity increases). It is important to recognize the impact of HTEs when making decisions about reducing sample size in anticipation of increased power due to covariate adjustment.</p>
</section>
<section id="a-hypothetical-example" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="a-hypothetical-example">A Hypothetical Example</h2>
<p>Suppose you are working on pre-specifying an adjusted analysis for an upcoming phase 2 trial. Assume that the targeted effect size is a <span class="math inline">\(25\%\)</span> reduction in mean decline in the treatment arm compared to the mean decline in the placebo arm for a progressive disease that has a continuous outcome measure. Furthermore, assume that you have developed a prognostic model using external/historical data and that the estimated correlation between the prognostic model predictions and the outcome was <span class="math inline">\(0.45\)</span> (<span class="math inline">\(r^2_{control}\approx0.2\)</span>). The following is an example of how to calculate ESSIs from covariate adjustment under two plausible treatment effect scenarios that are consistent with the targeted effect size:</p>
<div class="column-body-outset">
<table class="table tg">
<thead>
<tr>
<th class="tg-baqh">
Treatment Effect Scenario
</th>
<th class="tg-baqh">
<span class="math inline">\(r_{control}\)</span>
</th>
<th class="tg-baqh">
<span class="math inline">\(r_{active}\)</span>
</th>
<th class="tg-baqh">
ESSI
</th>
</tr>
</thead>
<tbody>
<tr>
<td class="tg-baqh">
Constant Absolute Delta<br><br><span class="math inline">\(E(Y_{active}|X)=E(Y_{control}|X)-\delta\)</span>
</td>
<td class="tg-baqh">
<span class="math inline">\(0.45\)</span>
</td>
<td class="tg-baqh">
<span style="font-weight:400;font-style:normal"><span class="math inline">\(0.45\)</span></span>
</td>
<td class="tg-baqh">
<span class="math inline">\(\Big(\frac{1}{1-r^2_{control}}-1\Big)*100\%=\)</span><br><span class="math inline">\(\Big(\frac{1}{1-{0.2}}-1\Big)*100\% =\)</span> <em><span class="math inline">\(25\%\)</span></em>
</td>
</tr>
<tr>
<td class="tg-baqh">
Constant Proportional Delta <br><br><span class="math inline">\(E(Y_{active}|X)=(1-\delta)E(Y_{control}|X)\)</span>
</td>
<td class="tg-baqh">
<span style="font-weight:400;font-style:normal"><span class="math inline">\(0.45\)</span></span>
</td>
<td class="tg-baqh">
<span class="math inline">\((1-\delta)r_{control} =\)</span><br><span class="math inline">\((1-.25)*0.45\)</span>
</td>
<td class="tg-baqh">
<span class="math inline">\(\Bigg(\frac{1}{1-r^2_{control} \; \; \big ( 1-\frac{\delta}{2}\big)^2}-1\Bigg)*100\%=\)</span><br><span class="math inline">\(\Bigg(\frac{1}{1-0.2\big(1-\frac{\delta}{2}\big)^2}-1\Bigg)*100\% =\)</span> <em><span class="math inline">\(18\%\)</span></em>
</td>
</tr>
<tr>
<td class="tg-baqh">
No Correlation in Treatment Arm
</td>
<td class="tg-baqh">
<span style="font-weight:400;font-style:normal"><span class="math inline">\(0.45\)</span></span>
</td>
<td class="tg-baqh">
<span class="math inline">\(0\)</span>
</td>
<td class="tg-baqh">
<span class="math inline">\(\Bigg(\frac{1}{1-\frac{r^2_{control}}{4}}-1\Bigg)*100\%=\)</span><br><span class="math inline">\(\Bigg(\frac{1}{1-\frac{0.2}{4}}-1\Bigg)*100\% =\)</span> <em><span class="math inline">\(5\%\)</span></em>
</td>
</tr>
</tbody>
</table>
</div>
<p>Assuming a constant absolute delta, the effective sample size increase from covariate adjustment is expected to be <span class="math inline">\(25\%\)</span>. However, <em>with a constant proportional treatment effect (which implies HTEs), the ESSI decreases to <span class="math inline">\(18\%\)</span></em>. A third, not entirely unrealistic worst-case scenario of no correlation between the covariate and outcome in the active treatment arm reduces the ESSI to just <span class="math inline">\(5\%\)</span>. The drop in ESSI with HTEs would be an important caveat to communicate if there is a temptation to decrease trial sample size.</p>
<p>All ESSI formulas assume a particular pattern of HTEs. Due to the risk of under-powering, if no prior data is available on HTEs, we do not routinely recommend reducing sample sizes for Phase 2 trials. Usually when designing a Phase 3 trial, there are data from earlier studies on HTEs that can be used to get direct estimates of HTEs and improve ESSI estimates. These improved ESSI estimates could be used to inform sample size reduction, if other trial requirements allow for it (e.g.&nbsp;safety and subgroup analyses). Whenever sample size is reduced, be sure to communicate these risks, consider the noise in ESSI estimates and study-to-study effects and tailor sample size reductions accordingly.</p>
</section>
<section id="application-to-geographic-atrophy-ga" class="level1">
<h1>Application to Geographic Atrophy (GA)</h1>
<p>Geographic atrophy (GA) is an advanced form of age-related macular degeneration (AMD) and is a major cause of blindness in developed countries worldwide. GA progression is assessed by GA lesion growth from baseline over time using Fundus Autofluorescence (FAF) images. This is usually the primary endpoint in clinical trials that evaluate treatment effects on progression.</p>
<div id="fig-ga-lesion-growth" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-lesion-growth-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_lesion_growth.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-lesion-growth-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: GA lesion progression over time on an FAF image
</figcaption>
</figure>
</div>
<p>Because GA progression is slow and varies between patients, clinical trials need to enroll many patients and follow them over a long period of time. Most recent GA trial designs followed patients for 12-18 months for the primary efficacy endpoint. However, the within patient noise of measuring GA progression over time is small and GA lesions progress linearly over the duration of the trial, making this a promising problem for a prediction model. This also allows us to use slopes calculated for each patient as the outcome for the prognostic modeling. Clinical trials often use change from baseline in GA area or slopes as outcomes, and sensitivity analyses showed that those are highly correlated and results are very similar for both. For simplicity, we will show the results for predicting the individual slopes.</p>
<div id="fig-ga-indi-patients" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-indi-patients-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_indiv_patients.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-indi-patients-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Examples of GA lesion growth for individual patients
</figcaption>
</figure>
</div>
<section id="prognostic-model-development" class="level2">
<h2 class="anchored" data-anchor-id="prognostic-model-development">Prognostic Model Development</h2>
<p>The modeling and data strategy was pre-specified and allowed a rigorous and fair comparison of a number of models with different levels of complexity and operational burden. The goal was to find the least complex model with the best performance.</p>
<p>Data from a previous GA development program was harmonized and divided into training, hold-out, and independent test sets as shown in the figure. The independent tests sets consisted of two studies that were entirely excluded from the training and hold-out evaluations. Within the training data set, models were trained using additional re-sampling techniques for parameter tuning and performance estimation.</p>
<div id="fig-ga-data-strategy" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-data-strategy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_data_strategy.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-data-strategy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Data Strategy for Model Development
</figcaption>
</figure>
</div>
<p>During training a model was selected from each of three model groups for hold-out and independent test set evaluation. These model groups were defined as:</p>
<ul>
<li>Model Group 1 (Benchmark): The first group of models was based on a set of standard, pre-specified features that are assessed at baseline. Various types of models were evaluated and compared, from a simple linear model to different types of machine learning models (e.g.&nbsp;support vector machines, gradient boosted machines, random forests, etc.). The features were:
<ul>
<li>Demographics: age, sex, smoking status</li>
<li>Measures of visual acuity: best corrected visual acuity (BCVA), low luminance visual acuity (LLVA), low luminance deficit (LLD = BCVA - LLVA)</li>
<li>Anatomical features assessed by a reading center: lesion size, lesion contiguity, lesion location, lesion distance to fovea, reticular pseudodrusen</li>
</ul></li>
<li>Model Group 2 (Benchmark + Run-in): This group additionally included a “run-in” estimate of growth rate, i.e.&nbsp;using the first 6 month to estimate the GA growth rate and predicting the future. Note that this required re-baselining the remaining outcome data to the 6 month time point. Due to the linear growth and low within patient noise, this is a promising approach but operationally complex to include in a trial as it requires high quality historical images from patients, or starting the trial with a pre-treatment run-in period, which may slow trial recruitment and delay the trial readout.</li>
<li>Model Group 3 (DL Models): The third group of models consists of end-to-end deep learning (DL) models from FAF and also OCT images at baseline. Three types of models were developed (each exploring different architectures): using FAF only, using OCT only and using FAF and OCT together <span class="citation" data-cites="DLpaper">(<a href="#ref-DLpaper" role="doc-biblioref">Anegondi et al. 2022</a>)</span>. The FAF only model would be the operationally preferred model.</li>
</ul>
<p>One model was selected from each model group using the relevant performance metrics (<span class="math inline">\(r^2\)</span> and ESSI) estimated using re-sampling techniques. The three selected models were then tested and compared on the holdout and independent test sets.</p>
</section>
<section id="results" class="level2">
<h2 class="anchored" data-anchor-id="results">Results</h2>
<p>The selected models were:</p>
<ul>
<li>Model Group 1 (Benchmark): A simple linear model with four features (lesion size, lesion contiguity, lesion distance to fovea and LLD) gave the best performance. More complex models like tree based models or added features did not add performance.</li>
<li>Model Group 2 (Benchmark + Run-in): A simple linear model with the four features from model group 1 and an estimate of growth rate as an additional fifth feature.</li>
<li>Model Group 3 (DL Models): A multi-task DL (CNN) model that uses a single FAF image as input to predict GA lesion size (same image) and GA growth rate <span class="citation" data-cites="DLpaper">(<a href="#ref-DLpaper" role="doc-biblioref">Anegondi et al. 2022</a>)</span>.</li>
</ul>
<p>The results were quite impressive; although operationally complex to implement, the run-in model appears to effectively increase the sample size by <span class="math inline">\(1/3\)</span> compared to an analysis using the benchmark model. If no DL FAF model were available, the additional complexity of the run-in might be justified by the <span class="math inline">\(ESSI\)</span>. However, the DL FAF model outperformed the benchmark and even the benchmark+run-in model in the hold-out as well as the two independent test sets <span class="citation" data-cites="DLpaper">(<a href="#ref-DLpaper" role="doc-biblioref">Anegondi et al. 2022</a>)</span>. This implies that the logistical complexities of the run-in model can be avoided with the DL FAF model, which is based on a single baseline time point and is still relatively simple to implement <span class="citation" data-cites="DLpaper">(<a href="#ref-DLpaper" role="doc-biblioref">Anegondi et al. 2022</a>)</span>. The table shows that the DL FAF model substantially increases the ESSI by at least 90% when compared to an unadjusted analysis and by 40%-80% compared to a simple adjustment using the known baseline features.</p>
<div id="fig-ga-model-results" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-model-results-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_model_results.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-model-results-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Results from selected model per group
</figcaption>
</figure>
</div>
</section>
<section id="model-in-action" class="level2">
<h2 class="anchored" data-anchor-id="model-in-action">Model in Action</h2>
<p>The MAHALO study independent test set shown in the previous section was the Phase 2 trial for Lampalizumab. There was an observed treatment effect of ∼20% slowing progression in the higher dose (LQ4 - monthly treatment) compared to the Sham (no treatment) arm. This effect was not confirmed by the two large global Phase 3 trials (SPECTRI and CHROMA), so in retrospect we know this was a false-positive result. When the decision was made to move forward with a Phase 3 trial, the DL FAF model was not available. However, it is an interesting post-hoc exercise to see how results from the Phase 2 trial would have changed had the DL FAF model been used for covariate adjustment.</p>
<p>The figure below shows the estimated mean changes in GA lesion size with and without adjusting for the FAF DL model. Covariate adjustment leads to better precision and hence tighter confidence intervals for the treatment effect estimate and also changes the estimate in case of imbalances. In this example, the estimate changes (from 20% to 6%) and becomes non-significant (a p-value threshold of 0.2 is usually used in a Phase 2 trial).</p>
<div id="fig-ga-mahalo-ex" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-mahalo-ex-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_mahalo_example.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-mahalo-ex-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: GA progression over time by treatment arm for unadjusted and adjusted analysis
</figcaption>
</figure>
</div>
<div id="fig-ga-mahalo-cis" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ga-mahalo-cis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/ga_mahalo_cis.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ga-mahalo-cis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Treatment effect estimates
</figcaption>
</figure>
</div>
<p>While there were many other considerations in the decision to move into Phase 3 trials (e.g.&nbsp;subgroup analyses), if results from the adjusted analysis had been available, Phase 3 decision making may have been different.</p>
</section>
</section>
<section id="brief-recap" class="level1">
<h1>Brief Recap</h1>
<p>Below is a brief outline summarizing how to get the most out of prognostic baseline variables via covariate adjustment:</p>
<ul>
<li>Analyze external/historical data
<ul>
<li>build a score/model</li>
<li>evaluate with ESSIs</li>
</ul></li>
<li>Determine model budget
<ul>
<li>use methods developed for prognostic modeling</li>
</ul></li>
<li>Spend the model budget wisely
<ul>
<li>choose the most prognostic factors among stratification and other prognostic variables</li>
<li>avoid dichotomania</li>
<li>by default, do not use interactions</li>
</ul></li>
<li>Estimate ATEs and TAMs
<ul>
<li>use standardized estimator</li>
<li>also works for non-continuous outcomes</li>
</ul></li>
<li>Inference for ATEs and TAMs
<ul>
<li>suggest SE estimates from TMLE literature</li>
<li>re-sampling methods are also an option</li>
</ul></li>
</ul>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>Covariate adjustment is a statistical free lunch. You can enjoy a rigorous, model-assisted analysis that increases precision and power, accounts for conditional bias and is supported by regulatory authorities at virtually no cost. With the growing availability of high-quality data for the identification of robust prognostic covariates, the benefits of an adjusted analysis will only continue to increase. We focused here on continuous endpoints, however the same standardized estimator and variance estimates can be applied to non-continuous outcomes (e.g.&nbsp;binary or ordinal) by substituting in the appropriate generalized linear “working” model and making minor adaptations in the standard error estimates <span class="citation" data-cites="TMLE2010">(<a href="#ref-TMLE2010" role="doc-biblioref">Rosenblum and Laan 2010</a>)</span>. ESSI formulas for non-continuous outcomes are a work-in-progress. While the concept of covariate adjustment has been around for decades, it seems to be underutilized or sub-optimally implemented (e.g.&nbsp;poor choice of covariates). We hope this guidance makes it clear how to rigorously implement covariate adjustment in randomized trials and encourages statisticians to take advantage of all of the benefits of adjusted analyses for randomized clinical trials.</p>
</section>
<section id="acknowledgements" class="level1">
<h1>Acknowledgements</h1>
<p><strong>Roche GA Development Team</strong>: Neha Anegondi, Lee Honigberg, Simon Gao, Qi Yang, Daniela Ferrara, Julia Cluceru, Verena Steffen</p>
<p><strong>Methods, Collaboration, Outreach (MCO)</strong>: Ray Lin, Marcel Wolbers</p>
<p><strong>Website Development</strong>: Doug Kelkhoff</p>
</section>
<section id="appendix" class="level1">
<h1>Appendix</h1>
<section id="essi-formulas-1" class="level2">
<h2 class="anchored" data-anchor-id="essi-formulas-1">ESSI Formulas</h2>
<ul>
<li><span class="math inline">\(Y\)</span> continuous outcome, <span class="math inline">\(X\)</span> baseline covariate</li>
<li><strong>Constant absolute <span class="math inline">\(\delta\)</span></strong>: <span class="math inline">\(E(Y_{active}|X)=E(Y_{control}|X)-\delta\)</span></li>
<li><strong>Constant proportional <span class="math inline">\(\delta\)</span></strong>: <span class="math inline">\(E(Y_{active}|X)=(1-\delta)E(Y_{control}|X)\)</span></li>
<li><span class="math inline">\(r_{control}\)</span> and <span class="math inline">\(r_{active}\)</span>: correlation between <span class="math inline">\(Y\)</span> and <span class="math inline">\(X\)</span> given control and active treatment, respectively.</li>
<li><span class="math inline">\(k=\frac{\sigma_{Y_{active}}}{\sigma_{Y_{control}}}\)</span></li>
<li><span class="math inline">\(\pi=Probability\;of\;active\;treatment\)</span>. Note, when <span class="math inline">\(\pi=1/2\)</span> the ESSI corresponding to the additive model will always equal the ESSI corresponding to the interaction model.</li>
</ul>
<div id="fig-essi-formulas" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-essi-formulas-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/essi_formulas.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-essi-formulas-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: ESSI Formulas
</figcaption>
</figure>
</div>
</section>
<section id="huber-white-sandwich-estimator" class="level2">
<h2 class="anchored" data-anchor-id="huber-white-sandwich-estimator">Huber White Sandwich Estimator</h2>
<p>Below is a simple simulation demonstrating how the Huber-White robust “sandwich” estimator will give standard error estimates that are too small when treatment effects are heterogeneous and a working regression model with treatment-by-covariate interactions is used.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sandwich)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">registerDoParallel</span>(<span class="at">cores=</span><span class="dv">4</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="do">################### SIMULATION STUDIES</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="do">####### Paradigm 1: Superpopulation</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample of size N from a population of covariates</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Each patient's treatment assignment is a Bernoulli(1/2) random variable</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Then generate a random outcome from the model using the covariate value </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># and treatment for all N patients</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">3</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>d  <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>var.eps0 <span class="ot">&lt;-</span> var.eps1 <span class="ot">&lt;-</span>  .<span class="dv">6</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>N <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>nsamp <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>p1.estimates <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="dv">1</span><span class="sc">:</span>nsamp,<span class="at">.combine =</span> rbind) <span class="sc">%dopar%</span> {</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  X.sim <span class="ot">&lt;-</span> <span class="fu">runif</span>(N,<span class="dv">0</span>,<span class="dv">1</span>) <span class="co"># sample covariate</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  X.sim.center <span class="ot">&lt;-</span> X.sim <span class="sc">-</span> <span class="fu">mean</span>(X.sim) <span class="co"># center covariate</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># outcome under control</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>  y0.sim  <span class="ot">&lt;-</span>  a<span class="sc">*</span>X.sim <span class="sc">+</span> b <span class="sc">+</span> <span class="fu">rnorm</span>(N,<span class="dv">0</span>,<span class="fu">sqrt</span>(var.eps0))</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># outcome under treatment</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  y1.sim  <span class="ot">&lt;-</span>   c<span class="sc">*</span>X.sim <span class="sc">+</span> d <span class="sc">+</span> <span class="fu">rnorm</span>(N,<span class="dv">0</span>,<span class="fu">sqrt</span>(var.eps1))</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  A.sim <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(N,<span class="dv">1</span>,p)</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>  Y.sim <span class="ot">&lt;-</span> y1.sim<span class="sc">*</span>A.sim <span class="sc">+</span> y0.sim<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>A.sim)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  df.sim <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Y.sim, A.sim, X.sim, X.sim.center) <span class="co"># A=0,1,2,...treatment arm factor</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Additive Model</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>  lm.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y.sim <span class="sc">~</span> A.sim <span class="sc">+</span> X.sim, <span class="at">data =</span> df.sim)</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>  lm.se <span class="ot">&lt;-</span> <span class="fu">summary</span>(lm.fit)[[<span class="dv">4</span>]][<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set treatment indicator to active treatment for all subjects</span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>  pred1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm.fit, <span class="at">newdata =</span> df.sim <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">A.sim=</span><span class="dv">1</span>))</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set treatment indicator to control for all subjects</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>  pred0 <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm.fit, <span class="at">newdata =</span> df.sim <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">A.sim=</span><span class="dv">0</span>)) </span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>  ate.add <span class="ot">&lt;-</span> <span class="fu">mean</span>(pred1) <span class="sc">-</span> <span class="fu">mean</span>(pred0)</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>  ate.add.se <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((A.sim<span class="sc">*</span>(Y.sim<span class="sc">-</span>pred1)<span class="sc">/</span><span class="fu">mean</span>(A.sim<span class="sc">==</span><span class="dv">1</span>) <span class="sc">+</span> pred1 <span class="sc">-</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mean</span>(pred1)<span class="sc">-</span>((<span class="dv">1</span><span class="sc">-</span>A.sim)<span class="sc">*</span>(Y.sim<span class="sc">-</span>pred0)<span class="sc">/</span><span class="fu">mean</span>(A.sim<span class="sc">==</span><span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>                               pred0 <span class="sc">-</span> <span class="fu">mean</span>(pred0)))<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="fu">sqrt</span>(N)</span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Interaction Model</span></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>  lm.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y.sim <span class="sc">~</span> A.sim<span class="sc">*</span>X.sim, <span class="at">data =</span> df.sim)</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set treatment indicator to active treatment for all subjects</span></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>  pred1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm.fit, <span class="at">newdata =</span> df.sim <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">A.sim=</span><span class="dv">1</span>))</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set treatment indicator to control for all subjects</span></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a>  pred0 <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm.fit, <span class="at">newdata =</span> df.sim <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">A.sim=</span><span class="dv">0</span>))</span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>  ate.int <span class="ot">&lt;-</span> <span class="fu">mean</span>(pred1) <span class="sc">-</span> <span class="fu">mean</span>(pred0)</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>  ate.int.se <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((A.sim<span class="sc">*</span>(Y.sim<span class="sc">-</span>pred1)<span class="sc">/</span><span class="fu">mean</span>(A.sim<span class="sc">==</span><span class="dv">1</span>) <span class="sc">+</span> pred1 <span class="sc">-</span> </span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">mean</span>(pred1)<span class="sc">-</span>((<span class="dv">1</span><span class="sc">-</span>A.sim)<span class="sc">*</span>(Y.sim<span class="sc">-</span>pred0)<span class="sc">/</span><span class="fu">mean</span>(A.sim<span class="sc">==</span><span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>                                            pred0 <span class="sc">-</span> <span class="fu">mean</span>(pred0)))<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="fu">sqrt</span>(N)</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>  <span class="do">## HW SEs</span></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>  lm.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y.sim <span class="sc">~</span> A.sim <span class="sc">+</span> X.sim, <span class="at">data =</span> df.sim)</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>  hw.add <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">vcovHC</span>(lm.fit, <span class="at">type=</span><span class="st">"HC0"</span>)[<span class="dv">2</span>,<span class="dv">2</span>])</span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>  lm.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y.sim <span class="sc">~</span> A.sim<span class="sc">*</span>X.sim.center, <span class="at">data =</span> df.sim)</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>  hw.int <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">vcovHC</span>(lm.fit, <span class="at">type=</span><span class="st">"HC0"</span>)[<span class="dv">2</span>,<span class="dv">2</span>]) </span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>  lm.fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Y.sim <span class="sc">~</span> A.sim<span class="sc">*</span>X.sim, <span class="at">data =</span> df.sim)</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a>  ate <span class="ot">&lt;-</span> <span class="fu">coefficients</span>(lm.fit)[<span class="dv">2</span>] <span class="sc">+</span> <span class="fu">coefficients</span>(lm.fit)[<span class="dv">4</span>]<span class="sc">*</span><span class="fu">mean</span>(df.sim<span class="sc">$</span>X.sim)</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>  myvcov <span class="ot">&lt;-</span> <span class="fu">vcovHC</span>(lm.fit, <span class="at">type=</span><span class="st">"HC0"</span>)</span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>  ate.sd <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>(df.sim<span class="sc">$</span>X.sim)<span class="sc">^</span><span class="dv">2</span><span class="sc">*</span>myvcov[<span class="dv">4</span>,<span class="dv">4</span>]<span class="sc">+</span>myvcov[<span class="dv">2</span>,<span class="dv">2</span>] <span class="sc">+</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">mean</span>(df.sim<span class="sc">$</span>X.sim)<span class="sc">*</span>myvcov[<span class="dv">2</span>,<span class="dv">4</span>])</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(ate.add, ate.int, ate.add.se, ate.int.se, hw.add, hw.int, lm.se))</span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a>p1.se.truth <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">apply</span>(p1.estimates[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="dv">2</span>,var))</span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a>tmle.mean.se <span class="ot">&lt;-</span> <span class="fu">apply</span>(p1.estimates[,<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>],<span class="dv">2</span>,mean)</span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a>hw.mean.se <span class="ot">&lt;-</span> <span class="fu">apply</span>(p1.estimates[,<span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>],<span class="dv">2</span>,mean)</span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a>t <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">data.frame</span>(<span class="at">truth =</span> p1.se.truth, <span class="at">tmle =</span> tmle.mean.se, <span class="at">HW =</span> hw.mean.se))</span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(t) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"True SE"</span>,<span class="st">"Mean TMLE SE"</span>,<span class="st">"Mean HW SE"</span>)</span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(t) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Additive Working Regression Model"</span>,<span class="st">"Interaction Working Regression Model"</span>)</span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(t, <span class="at">align =</span> <span class="st">"cc"</span>, <span class="at">caption =</span> <span class="st">"Superpopulation Sampling Framework"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Superpopulation Sampling Framework</caption>
<colgroup>
<col style="width: 15%">
<col style="width: 40%">
<col style="width: 44%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Additive Working Regression Model</th>
<th style="text-align: center;">Interaction Working Regression Model</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">True SE</td>
<td style="text-align: center;">0.0214373</td>
<td style="text-align: center;">0.0214312</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean TMLE SE</td>
<td style="text-align: center;">0.0211753</td>
<td style="text-align: center;">0.0211741</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mean HW SE</td>
<td style="text-align: center;">0.0211780</td>
<td style="text-align: center;">0.0154906</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="references" class="level1">




</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-DLpaper" class="csl-entry" role="listitem">
Anegondi, Neha, Simon S. Gao, Verena Steffen, Richard F. Spaide, SriniVas R. Sadda, Frank G. Holz, Christina Rabe, et al. 2022. <span>“Deep Learning to Predict Geographic Atrophy Area and Growth Rate from Multi-Modal Imaging.”</span> <em>Ophthalmology Retina</em>. <a href="https://doi.org/10.1016/j.oret.2022.08.018">https://doi.org/10.1016/j.oret.2022.08.018</a>.
</div>
<div id="ref-covid" class="csl-entry" role="listitem">
Benkeser, David, Ivan Diaz, Alex Luedtke, Jodi Segal, Daniel Scharfstein, and Michael Rosenblum. 2020. <span>“Improving Precision and Power in Randomized Trials for COVID-19 Treatments Using Covariate Adjustment, for Binary, Ordinal, and Time-to-Event Outcomes.”</span> <em>Biometrics</em>.
</div>
<div id="ref-Colantuoni2015" class="csl-entry" role="listitem">
Colantuoni, Elizabeth, and Michael Rosenblum. 2015. <span>“Leveraging Prognostic Baseline Variables to Gain Precision in Randomized Trials.”</span> <em>Stat. Med</em>.
</div>
<div id="ref-Daniel" class="csl-entry" role="listitem">
Daniel, Rhian, Jingjing Zhang, and Daniel Farewell. 2020. <span>“Making Apples from Oranges: Comparing Noncollapsible Effect Estimators and Their Standard Errors After Adjustment for Different Covariate Sets.”</span> <em>Biometrical Journal</em>.<a href=" https://doi.org/10.1002/bimj.201900297"> https://doi.org/10.1002/bimj.201900297</a>.
</div>
<div id="ref-CUPED" class="csl-entry" role="listitem">
Deng, Alex, Ya Xu, Ron Kohavi, and Toby Walker. 2013. <span>“Improving the Sensitivity of Online Controlled Experiments by Utilizing Pre-Experiment Data.”</span> <em>WSDM</em>.
</div>
<div id="ref-Peng" class="csl-entry" role="listitem">
Ding, Peng, Xinran Li, and Luke Miratrix. 2017. <span>“Bridging Finite and Super Population Causal Inference.”</span> <em>J. Causal Infer.</em>
</div>
<div id="ref-Ding" class="csl-entry" role="listitem">
———. 2019. <span>“Rerandomization and Regression Adjustment.”</span> <em>arXiv</em>. <a href="https://arxiv.org/abs/1906.11291">https://arxiv.org/abs/1906.11291</a>.
</div>
<div id="ref-FDA" class="csl-entry" role="listitem">
FDA, U. S. 2021. <span>“Adjusting for Covariates in Randomized Clinical Trials for Drugs and Biological Products Guidance for Industry.”</span> <a href="https://www.fda.gov/regulatory-information/search-fda-guidance-documents/adjusting-covariates-randomized-clinical-trials-drugs-and-biological-products">https://www.fda.gov/regulatory-information/search-fda-guidance-documents/adjusting-covariates-randomized-clinical-trials-drugs-and-biological-products</a>.
</div>
<div id="ref-Freedman2008a" class="csl-entry" role="listitem">
Freedman, D. A. 2008a. <span>“Randomization Does Not Justify Logistic Regression.”</span> <em>Statistical Science</em>, 2008a.
</div>
<div id="ref-Freedman2008b" class="csl-entry" role="listitem">
Freedman, David A. 2008b. <span>“ON REGRESSION ADJUSTMENTS IN EXPERIMENTS WITH SEVERAL TREATMENTS.”</span> <em>The Annals of Applied Statistics</em>, 2008b.
</div>
<div id="ref-Ge2011" class="csl-entry" role="listitem">
Ge, Miaomiao, L. Kathryn Durham, R. Daniel Meyer, Wangang Xie, and Neal Thomas. 2011. <span>“Covariate-Adjusted Difference in Proportions from Clinical Trials Using Logistic Regression and Weighted Risk Differences.”</span> <em>Drug Information Journal</em>.
</div>
<div id="ref-MLRATE" class="csl-entry" role="listitem">
Guo, Yongyi, Dominic Coey, Mikael Konutgan an Wenting Li, Chris Schoener, and Matt Goldman. 2022. <span>“Machine Learning for Variance Reduction in Online Experiments.”</span> <em>arXiv</em>.
</div>
<div id="ref-rms" class="csl-entry" role="listitem">
Harrell, Frank E. 2011. <em>Regression Modeling Strategies</em>. New York: Springer.
</div>
<div id="ref-imbens" class="csl-entry" role="listitem">
Imbens, Guido, and Jeffrey Wooldridge. 2009. <span>“Recent Developments in the Econometrics of Program Evaluation.”</span> <em>J. Econ. Lit</em>.
</div>
<div id="ref-CUPAC" class="csl-entry" role="listitem">
Li, Jeff, Yixin Tang, and Jared Bauman. 2020. <span>“Improving Experimental Power Through Control Using Predictions as Covariates.”</span> <a href="https://doordash.engineering/2020/06/08/improving-experimental-power-through-control-using-predictions-as-covariate-cupac/" class="uri">https://doordash.engineering/2020/06/08/improving-experimental-power-through-control-using-predictions-as-covariate-cupac/</a>.
</div>
<div id="ref-lin2013" class="csl-entry" role="listitem">
Lin, Winston. 2013. <span>“Agnsotic Notes on Regression Adjustments to Experimental Data: Reexamining Freedman’s Critique.”</span> <em>The Annals of Applied Statistics</em>.
</div>
<div id="ref-Riley2019" class="csl-entry" role="listitem">
Riley, Richard D, Kym Snell, Joie Ensor, Danielle Burke, Frank Harrell, Karel Moons, and Gary Collins. 2019. <span>“Minimum Sample Size for Developing a Multivariable Prediction Model: Part i - Continuous Outcomes.”</span> <em>Stat. Med.</em>
</div>
<div id="ref-Riley2020" class="csl-entry" role="listitem">
Riley, Richard, Joie Ensor, Kym Snell, Frank Harrell, Glen Martin, Johannes Reitsma associate, Karel Moons, Gary Collins, and Maarten van Smeden. 2020. <span>“Calculating the Sample Size Required for Developing a Clinical Prediction Model.”</span> <em>BMJ</em>.
</div>
<div id="ref-ML" class="csl-entry" role="listitem">
Rosenblum, Michael. 2020. <span>“Machine Learning for Leveraging Prognostic Baseline Variables to Gain Precision and Reduce Sample Size in Randomized Trials.”</span> <span class="smallcaps">url:</span>&nbsp;<a href="http://rosenblum.jhu.edu/" class="uri">http://rosenblum.jhu.edu/</a>.
</div>
<div id="ref-TMLE2010" class="csl-entry" role="listitem">
Rosenblum, Michael, and Mark J. van der Laan. 2010. <span>“Simple, Efficient Estimators of Treatment Effects in Randomized Trials Using Generalized Linear Models to Leverage Baseline Variables.”</span> <em>The International Journal of Biostatistics</em>.
</div>
<div id="ref-Rosenblum2009" class="csl-entry" role="listitem">
Rosenblum, Michael, and Mark van der Laan. 2009. <span>“Using Regression Models to Analyze Randomized Trials: Asymptotically Valid Hypothesis Tests Despite Incorrectly Specified Models.”</span> <em>Biometrics</em>.
</div>
<div id="ref-Steingrimsson2016" class="csl-entry" role="listitem">
Rosenblum, Michael, and Jon Steingrimsson. 2016. <span>“Matching the Efficiency Gains of the Logistic Regression Estimator While Avoiding Its Interpretability Problems, in Randomized Trials.”</span> <em>Johns Hopkins University, Dept. Of Biostatistics Working Papers.</em>
</div>
<div id="ref-Rudin" class="csl-entry" role="listitem">
Rudin, Cynthia. 2019. <span>“Stop Explaining Black Box Machine Learning Models for High Stakes Decisions and Use Interpretable Models Instead.”</span> <em>Nat Mach Intell</em>. <a href="https://doi.org/10.1038/s42256-019-0048-x">https://doi.org/10.1038/s42256-019-0048-x</a>.
</div>
<div id="ref-unlearn" class="csl-entry" role="listitem">
Schuler, Alejandro, David Walsh, Diana Hall, Jon Walsh, and Charles Fisher. 2020. <span>“Increasing the Efficiency of Randomized Trial Estimates via Linear Adjustment for a Prognostic Score.”</span> <em>arXiv</em>.
</div>
<div id="ref-Senn" class="csl-entry" role="listitem">
Senn, S. J. 1989. <span>“Covariate Imbalance and Random Allocation in Clinical Trials.”</span> <em>Statistics in Medicine</em>.
</div>
<div id="ref-Senn2010" class="csl-entry" role="listitem">
Senn, Stephen. 2010. <span>“Some Considerations Concerning Covariates in Clinical Trials.”</span> <span class="smallcaps">url:</span>&nbsp;<a href="https://www.page-meeting.org/pdf_assets/9236-SennCovariates.pdf" class="uri">https://www.page-meeting.org/pdf_assets/9236-SennCovariates.pdf</a>.
</div>
<div id="ref-Dichotomania" class="csl-entry" role="listitem">
Senn, Stephen J. 2005. <span>“Dichotomania: An Obsessive Compulsive Disorder That Is Badly Affecting the Quality of Analysis of Pharmaceutical Trials.”</span> In.
</div>
<div id="ref-Tian2012" class="csl-entry" role="listitem">
Tian, Lu, Tianxi Cai, Lihui Zhao, and Lee-Jen Wei. 2012. <span>“On the Covariate-Adjusted Estimation for an Overall Treatment Difference with Data from a Randomized Comparative Clinical Trial.”</span> <em>Biostatistics</em>.
</div>
<div id="ref-owkin" class="csl-entry" role="listitem">
Trower, Antonia, Felix Balazard, and Radha Patel. 2020. <span>“Leveraging Machine Learning to Optimize Clinical Trials: The Power of Covariate Adjustment.”</span> <a href="https://owkin.com/publications-and-news/blogs/leveraging-machine-learning-to-optimize-clinical-trials-the-power-of-covariate-adjustment">https://owkin.com/publications-and-news/blogs/leveraging-machine-learning-to-optimize-clinical-trials-the-power-of-covariate-adjustment</a>.
</div>
<div id="ref-Tsiatis2008" class="csl-entry" role="listitem">
Tsiatis, Anastasios A., Marie Davidian, Min Zhang, and Xiaomin Lu. 2008. <span>“Covariate Adjustment for Two-Sample Treatment Comparisons in Randomized Clinical Trials: A Principled yet Flexible Approach.”</span> <em>Stat. Med.</em>
</div>
<div id="ref-Wang2021" class="csl-entry" role="listitem">
Wang, Bingkai, Ryoko Susukida, Ramin Mojtabai, Masoumeh Amin-Esmaeili, and Michael Rosenblum. 2021. <span>“Model-Robust Inference for Clinical Trials That Improve Precision by Stratified Randomization and Covariate Adjustment.”</span> <em>Journal of the American Statistical Association</em>.
</div>
<div id="ref-covid2" class="csl-entry" role="listitem">
Williams, Nicholas, Michael Rosenblum, and Ivan Diaz. 2021. <span>“Optimizing Precision and Power by Machine Learning in Randomized Trials, with an Application to COVID-19.”</span> <em>arXiv Preprint arXiv:2109.04294.</em>
</div>
<div id="ref-jackknife" class="csl-entry" role="listitem">
Wolbers, Marcel, Alessandro Noci, Paul Delmar, Craig Gower-Page, Sean Yiu, and Jonathan W. Bartlett. 2022. <span>“Standard and Reference-Based Conditional Mean Imputation.”</span> <em>Pharmaceutical Statistics</em>. <a href="https://doi.org/10.1002/pst.2234">https://doi.org/10.1002/pst.2234</a>.
</div>
<div id="ref-Netflix" class="csl-entry" role="listitem">
Wong, Jeffrey, Randall Lewis, and Matthew Wardrop. 2019. <span>“Efficient Computation of Linear Model Treatment Effects in an Experimentation Platform.”</span> <em>arXiv</em>.
</div>
<div id="ref-Tsiatis2001" class="csl-entry" role="listitem">
Yang, Li, and Anastasios A. Tsiatis. 2001. <span>“Efficiency Study of Estimators in a Pretest-Posttest Trial.”</span> <em>The American Statistician</em>.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@misc{schiffman;christina rabe;michel friesenhahn2022,
  author = {Schiffman;Christina Rabe;Michel Friesenhahn, Courtney},
  title = {How to Get the Most Out of Prognostic Baseline Variables in
    Clinical Trials. {Effective} Strategies for Employing Covariate
    Adjustment and Stratification.},
  date = {2022-10-12},
  url = {stats4datascience.com},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-schiffman;christina rabe;michel friesenhahn2022" class="csl-entry quarto-appendix-citeas" role="listitem">
Schiffman;Christina Rabe;Michel Friesenhahn, Courtney. 2022. <span>“How
to Get the Most Out of Prognostic Baseline Variables in Clinical Trials.
Effective Strategies for Employing Covariate Adjustment and
Stratification.”</span> <a href="https://stats4datascience.com">stats4datascience.com</a>.
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>